### ISSUE TITLE
Using an MacOS and Chrome (Gmail) doesn't attach the file

### ISSUE THREAD

**Issue by farzadmf** [nik012003/ripdrag#19]
Hi there,

This project is really what I needed on my Mac, and I was super excited about it, and I installed it immediately.

Tried to test it to attach a file in browser's Gmail (using Chrome), and when I drag the file, it basically pastes the path to the file, instead of attaching it to the email.

If I do the same using Finder (MacOS's file manager), it properly attaches the file.

I was wondering if this is something that is a known issue/can be fixed.

Thank you

**Comment by nik012003** [comment]
Yes, I am aware of this. Unfortunately, I confirmed that this is an issue with gtk4 iteself in both MacOS and Windows systems. As soon as the gtk fixes it, it should work automatically on ripdrag.

**Comment by nik012003** [comment]
I didn't see one open unfortunately, plese open one here https://gitlab.gnome.org/GNOME/gtk/-/issues if you feel like it.
I will try to fill the issue up with more information once I'll go back to macos... In the meanwhile I pushed some of the tests I was doing with on the macos-tests branch

**Comment by xxyxxyxyx1** [comment]
Is there an open issue GTK4 issue to follow?
I do not know what the problem is with GTK4, so I do not know what to search on their issue tracker.

**Comment by nik012003** [comment]
> Is there an open issue GTK4 issue to follow? I do not know what the problem is with GTK4, so I do not know what to search on their issue tracker.

I don't think there is one. I don't have a working macos install to reproduce, but you should be able to repro and report the bug by compiling gtk4 demos and testing drag and drop there.

**Comment by ilyagr** [comment]
I ended up using https://eternalstorms.at/yoink/mac/ for now. It's a little too elaborate for my liking, is not open-source, and has a nag screen until you pay, but works.

**Comment by Mellbourn** [comment]
@ilyagr that "Tip number 5", i.e. copying a file to clipboard in the terminal, can be achieved with a free script, e.g. 
```
copy(){ osascript -e{'on run{a}','set the clipboard to posix file a',end} "$(greadlink -f -- "$1")";}
```
(source: https://apple.stackexchange.com/a/15484/12219)

**Comment by ilyagr** [comment]
Actually, I think I've seen the light. You don't need any app on a Mac to do this. You don't even need a script.

All you need is `open -a TextEdit anyfile`, and then use the proxy icon at the top of the screen to drag the file. By default, you need to click on the filename in the title for the icon to show up.

<img width="763" alt="SCR-20241001-rkni" src="https://github.com/user-attachments/assets/2c79e2c8-21ae-4a70-99a5-14992f7bf20b">

I know it works since this is how the image above was attached to this issue :)

Obviously, TextEdit is not the only app that can do this, but it's one of the apps that seems to be willing to open any file.

You can ask MacOS to always show the proxy icons with the "Show window title icons" setting.

<img width="827" alt="image" src="https://github.com/user-attachments/assets/00398b0d-0cd7-4c4b-b4a8-0b95b5d854b8">

**Comment by ilyagr** [comment]
> @ilyagr that "Tip number 5", i.e. copying a file to clipboard in the terminal, can be achieved with a free script, e.g.
> 
> ```
> copy(){ osascript -e{'on run{a}','set the clipboard to posix file a',end} "$(greadlink -f -- "$1")";}
> ```
> 
> (source: https://apple.stackexchange.com/a/15484/12219)

Thanks, this works for me so far as well (though it does a slightly different thing).

By the way, on recent versions of Mac OS (I'm on Sonoma, but I think this has been the case for some time), `/usr/bin/readlink` supports the `-f` option, so there is probably no need for `greadlink`. Just replace it with `readlink`.

**Comment by xxyxxyxyx1** [comment]
@ilyagr, thank you very much for the window title icon idea! At first, I could not get it to work, but I think I have figured out the trick:
you must click and hold the mouse stationary on the icon for some small amount of time before moving the mouse. Even if the window bar has already been clicked, clicking the icon and immediately dragging just moves the window instead. 

Two more notes:
- according to the manpage, `open -e` will open the file in TextEdit. Slightly shorter than `open -a TextEdit` ;)
- it would be nice if this could be achieved while opening the file in read-only-mode, like `vim -m`, to prevent accidental editing. I couldn't figure it out, though.

**Comment by ilyagr** [comment]
> according to the manpage, open -e will open the file in TextEdit. Slightly shorter than open -a TextEdit ;)

Nice!

> it would be nice if this could be achieved while opening the file in read-only-mode, like vim -m, to prevent accidental editing. I couldn't figure it out, though.

Nice idea! For Vim fans, you can install https://github.com/macvim-dev/macvim, set up the `mvim` shortcut in your PATH, and then use `mvim -M`. There's currently a minor bug, https://github.com/macvim-dev/macvim/issues/1503, though, that makes the proxy icon invisible. It's still there, though.

**Comment by xxyxxyxyx1** [comment]
I have encountered a situation where opening with the TextEdit does not work: zip archives. 
However, your new suggestion of `mvim -M` does work. Thanks yet again.

**Comment by xxyxxyxyx1** [comment]
Do you have a suggestion for doing this with many files? 
For example, if I have many zip archives to drag onto a webpage.

**Comment by ThSGM** [comment]
Since ripdrag is probably the most promising new development of the classic dragon, I wanted to just make some notes now in Jan 2025. 

I'm wondering if someone has managed to source an alternative or get ripdrag working for Mac?

I did manage to locate this one: https://github.com/Wevah/dragterm/issues/2

and it does work, albeit with problems being a non-exiting process. Ripdrag looks much more promising, but I'm confused at its current status for Mac. I used the pre-built version located here: https://github.com/Wevah/dragterm/issues/2#issuecomment-2081437994

Moreover, since there might be a non-zero intersection between those using command-line file browsers, I wanted to note that I managed to get a the above drag.zip working perfectly with [yazi](https://github.com/sxyazi/yazi) using the comment here: https://github.com/sxyazi/yazi/discussions/327

```
{ on = [ "<C-n>" ], run = '''
    shell 'drag "$@" -x 2>/dev/null &' --confirm
''' },

```
